org.eclipse.xtext.xbase.ui/src/org/eclipse/xtext/xbase/ui/validation/XbaseUIValidator.java

```diff
@@ -222,14 +228,38 @@ protected IClasspathEntry getResolvedClasspathEntry(IJavaProject javaProject, /*
 		IClasspathEntry result = null;
 		JavaProject castedProject = (JavaProject) javaProject;
 		castedProject.getResolvedClasspath(); // force the resolved entry cache to be populated
-		@SuppressWarnings("rawtypes")
-		Map rootPathToResolvedEntries = castedProject.getPerProjectInfo().rootPathToResolvedEntries;
+		Map<IPath, IClasspathEntry> rootPathToResolvedEntries = getRootPathToResolvedEntries(castedProject.getPerProjectInfo());
 		if (rootPathToResolvedEntries != null) {
-			result = (IClasspathEntry) rootPathToResolvedEntries.get(root.getPath());
+			result = rootPathToResolvedEntries.get(root.getPath());
 			if (result == null)
-				result = (IClasspathEntry) rootPathToResolvedEntries.get(root.getJavaProject().getPath());
+				result = rootPathToResolvedEntries.get(root.getJavaProject().getPath());
 		}
 		
 		return result;
 	}
+	
+	
+	protected final boolean isJdtCoreVersionAtLeast3390 = JavaCore.getPlugin().getBundle().getVersion().compareTo(new Version(3, 39, 0)) >= 0;
+	protected Map<IPath, IClasspathEntry> getRootPathToResolvedEntries(PerProjectInfo info) {
+		if (isJdtCoreVersionAtLeast3390) {
+			try {
+				Method m = PerProjectInfo.class.getDeclaredMethod("getRootPathToResolvedEntries");
```

Comment:
Nit:
We could you MethodHandles.findGetter and MethodHandles.findVirtual to obtain a MethodHandle once and use that throughout the lifecyle of this class.

Reply:
am not familiar with that api

Reply:
https://github.com/eclipse/xtext/pull/3128
something like this?